src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -22,36 +22,35 @@ public class DatadogHttpClient : IDatadogClient
         private readonly string _url;
         private readonly DatadogLogRenderer _renderer;
         private readonly HttpClient _client;
-
-        /// <summary>
-        /// Max number of retries when sending failed.
-        /// </summary>
-        private const int MaxRetries = 10;
+        private readonly int _maxRetries;
 
         /// <summary>
         /// Max backoff used when sending failed.
         /// </summary>
         private const int MaxBackoff = 30;
 
-        public DatadogHttpClient(string url, DatadogLogRenderer renderer, HttpClient client)
+        public DatadogHttpClient(string url, DatadogLogRenderer renderer, HttpClient client, int maxRetries)
         {
             _url = url;
             _renderer = renderer;
             _client = client;
+            _maxRetries = maxRetries;
         }
 
-        public Task WriteAsync(IEnumerable<LogEvent> events)
+        public Task WriteAsync(IReadOnlyCollection<LogEvent> events)
         {
             var builtEvents = BuildEvents(events);
             var tasks = builtEvents.Select(post => Post(post));
             return Task.WhenAll(tasks);
         }
 
-        private List<JsonPayloadBuilder> BuildEvents(IEnumerable<LogEvent> events)
+        private List<JsonPayloadBuilder> BuildEvents(IReadOnlyCollection<LogEvent> events)
         {
             var builders = new List<JsonPayloadBuilder>();
             var builder = new JsonPayloadBuilder();
 
+            var chunkBuffer = new List<string>(events.Count());
+            var logEvents = new List<LogEvent>(events.Count());
```

Comment:
```suggestion
```

These are not being used anywhere. The change from `IEnumerable` to `IReadOnlyCollection` does not seem necessary.